### DISCUSSION TITLE
How to handle one time operations like CRDs update for operators?

### DISCUSSION THREAD

**Discussion by ksawerykarwacki** [fluxcd/flux2#2554] [Q&A] [answered]
Let's say I have bunch of operators deployed. When I upgrade them I have to manually apply CRD after the migration (e.g. https://strimzi.io/docs/operators/in-development/full/deploying.html#con-upgrade-sequence-options-str) 

But in some cases there is not versioned package (like Strimzi above) but just file from master branch (https://raw.githubusercontent.com/Altinity/clickhouse-operator/master/deploy/operator/parts/crd.yaml). 

I don't want to always sync it because in some scenarios this would override my CRDs before I will update operator itself. 

How to handle that kind of scenarios properly with GitOps?

**Comment by kingdonb** [comment, marked as answer]
If you have some CRDs that you only want to apply most of the time, and cease syncing them while other things continue syncing (for example, the migration/upgrade you described) the best way to handle this in GitOps is to put the CRDs in a Flux Kustomization by themselves and suspend it for the duration of the migration, then update the spec to point at the new version which you're ready to sync, then `flux resume` the suspended Kustomization and let it go back to maintaining the resource with your pinned version.

You might also need multiple GitRepository resources in order to facilitate this workflow, since one part needs to be at a different version than the other part for a brief span of time. It could also work with just one GitRepository if you have understood which parts are doing what, and are using the suspend workflow in the right order.
